hopsworks-common/src/main/java/io/hops/hopsworks/common/project/ProjectController.java

```diff
@@ -1401,6 +1399,21 @@ private void waitForJobLogs(List<ApplicationReport> projectsApps, YarnClient cli
       }
     }
   }
+  
+  private void removeCertificatesFromMaterializer(Project project, Collection<ProjectTeam> teams) {
+    for (ProjectTeam team : teams) {
+      certificateMaterializer.forceRemoveLocalMaterial(team.getUser().getUsername(), project.getName(), null, true);
+      String remoteCertsDirectory = settings.getHdfsTmpCertDir() + Path.SEPARATOR +
+          hdfsUsersController.getHdfsUserName(project, team.getUser());
+      if (remoteCertsDirectory.equals(settings.getHdfsTmpCertDir())) {
```

Comment:
@kouzant shouldn't this be covered by a test, instead of going through the clause for every invocation?

Reply:
It is a safety net called when we delete a Project. Normally, each service individually that requested a crypto material should remove it, but if for some reason it doesn't this snippet will make sure there is no garbage left behind.

Reply:
Thanks, please add a comment in the code then